src/module/utils/TwodsixDiceRoll.ts

```diff
@@ -0,0 +1,170 @@
+import {Crit} from "../../types/twodsix";
+import { TWODSIX } from "../config";
+import TwodsixActor from "../entities/TwodsixActor";
+import TwodsixItem from "../entities/TwodsixItem";
+import { advantageDisadvantageTerm } from "../i18n";
+import { getKeyByValue } from "./sheetUtils";
+import { TwodsixRollSettings } from "./TwodsixRollSettings";
+
+export class TwodsixDiceRoll {
```

Comment:
This class holds a twodsix skill/characteristic roll object, which is aware about effect, target, etc. It also can send a chat message about the result.